web/crux/src/app/template/template.http.controller.ts

```diff
@@ -0,0 +1,55 @@
+import { Body, Controller, Get, Response, Param, Post, Header, UseGuards, UseInterceptors } from '@nestjs/common'
+import { ApiBody, ApiCreatedResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger'
+import { Identity } from '@ory/kratos-client'
+import HttpLoggerInterceptor from 'src/interceptors/http.logger.interceptor'
+import PrismaErrorInterceptor from 'src/interceptors/prisma-error-interceptor'
+import TemplateFileService from 'src/services/template.file.service'
+import { Response as ExpressResponse } from 'express'
+import { HttpCode } from '@nestjs/common/decorators'
+import { CreatedResponse, CreatedWithLocation } from '../shared/created-with-location.decorator'
+import CreatedWithLocationInterceptor from '../shared/created-with-location.interceptor'
+import JwtAuthGuard, { IdentityFromRequest } from '../token/jwt-auth.guard'
+import { CreateProductFromTemplateDto, TemplateDto } from './template.dto'
+import TemplateService from './template.service'
+import { ProductDto } from '../product/product.dto'
+
+const ROUTE_TEMPLATES = 'templates'
+const ROUTE_TEMPLATE_ID = ':templateId'
+const TemplateId = () => Param('templateId')
+
+@Controller(ROUTE_TEMPLATES)
+@ApiTags(ROUTE_TEMPLATES)
+@UseGuards(JwtAuthGuard)
+@UseInterceptors(HttpLoggerInterceptor, PrismaErrorInterceptor, CreatedWithLocationInterceptor)
+export default class TemplateHttpController {
+  constructor(private service: TemplateService, private templateFileService: TemplateFileService) {}
+
+  @Get()
+  @ApiOkResponse({ type: TemplateDto, isArray: true })
+  async getTemplates(): Promise<TemplateDto[]> {
+    return this.templateFileService.getTemplates()
+  }
+
+  @Post()
+  @HttpCode(204)
```

Comment:
This is not right. The http code should be 201. However the `@CreatedWithLocation()` annotation already provides that.

Reply:
Fixed.